pkg/types/builder.go

```diff
@@ -31,8 +32,9 @@ type Generated struct {
 	Types    []*types.Named
 	Comments twtypes.Comments
 
-	ForProviderType *types.Named
-	AtProviderType  *types.Named
+	ForProviderType  *types.Named
+	InitProviderType *types.Named
```

Comment:
Not for this PR, but should we consider putting the generation of the `spec.initProvider` API behind a configuration parameter (make the generation of the API optional)?

How do we feel about the following question? Will all the providers generated by upjet need this API on their MRs? From our previous discussions, my understanding is that not every MR (or provider in the higher level) may need this API. If this is the case, then we can consider opening an issue in upjet so that we can parameterize the code generation pipeline.

Reply:
I would say that probably because of tags it will be needed for all resources. But as an alpha feature we can see how it is used. We could add a configuration parameter not to generate initProviders for some fields, and just put an empty struct for.